Jenkinsfile

```diff
@@ -18,315 +18,326 @@ pipeline {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-dark-ingestion.xml ' +
-								'-m dark_ingestion /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+				    withKubeConfig([credentialsId: "prod-kube-config"]) {
+					    sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+						        "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+						        "--junitxml=/nres/code/pytest-dark-ingestion.xml -m dark_ingestion /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-dark-ingestion.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						    sh("kubectl -n dev cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-dark-ingestion.xml " +
+						            "pytest-dark-ingestion.xml")
+						    junit "pytest-dark-ingestion.xml"
 						}
-						junit 'pytest-dark-ingestion.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Master-Dark-Creation') {
+		stage("Test-Master-Dark-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-master-dark.xml ' +
-								'-m master_dark /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+			        withKubeConfig([credentialsId: "prod-kube-config"]) {
+						sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+						        "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+						        "--junitxml=/nres/code/pytest-master-dark.xml -m master_dark /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-master-dark.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						    sh("kubectl -n dev cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-master-dark.xml pytest-master-dark.xml")
+						    junit "pytest-master-dark.xml"
 						}
-						junit 'pytest-master-dark.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Flat-Ingestion') {
+		stage("Test-Flat-Ingestion") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-flat-ingestion.xml ' +
-								'-m flat_ingestion /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+				    withKubeConfig([credentialsId: "prod-kube-config"]) {
```

Comment:
If all these steps use the `dev` namespace, you can use the `dev-kube-config` credentials.